bigbluebutton-html5/imports/ui/components/presentation/component.jsx

```diff
@@ -1,78 +1,246 @@
+      let clientHeight;
+      let clientWidth;
+      if (this.props.userIsPresenter) {
+        clientHeight = whiteboardSizeAvailable.clientHeight;
+        clientWidth = whiteboardSizeAvailable.clientWidth;
+      // user is not a presenter - we can get the sizes of the presentationPaper
+      // direct parent of the svg wrapper
+      } else {
+        clientHeight = presentationPaper.clientHeight;
+        clientWidth = presentationPaper.clientWidth;
+      }
+
+      paperSizes.paperHeight = clientHeight;
+      paperSizes.paperWidth = clientWidth;
+    }
+    return paperSizes;
+  }
+
+  getInitialPaperSizes() {
+    // determining the paperWidth and paperHeight (available space for the svg) on the initial load
+    const paperSizes = this.getPaperSizes();
+    if (Object.keys(paperSizes).length > 0) {
+      // setting the state of the paperWidth and paperHeight (available space for the svg)
+      // and set the showSlide to true to start rendering the slide
+      paperSizes.showSlide = true;
+      this.setState(paperSizes);
+    }
+  }
+
+  handleResize() {
+    const paperSizes = this.getPaperSizes();
+    if (Object.keys(paperSizes).length > 0) {
+      // updating the size of the space available for the slide
+      this.setState(paperSizes);
+    }
+  }
+
+  calculateSize() {
+    const originalWidth = this.props.currentSlide.width;
+    const originalHeight = this.props.currentSlide.height;
+
+    let adjustedWidth;
+    let adjustedHeight;
+
+    // Slide has a portrait orientation
+    if (originalWidth <= originalHeight) {
+      adjustedWidth = (this.state.paperHeight * originalWidth) / originalHeight;
+      if (this.state.paperWidth < adjustedWidth) {
+        adjustedHeight = (this.state.paperHeight * this.state.paperWidth) / adjustedWidth;
+        adjustedWidth = this.state.paperWidth;
+      } else {
+        adjustedHeight = this.state.paperHeight;
+      }
+
+      // Slide has a landscape orientation
+    } else {
+      adjustedHeight = (this.state.paperWidth * originalHeight) / originalWidth;
+      if (this.state.paperHeight < adjustedHeight) {
+        adjustedWidth = (this.state.paperWidth * this.state.paperHeight) / adjustedHeight;
+        adjustedHeight = this.state.paperHeight;
+      } else {
+        adjustedWidth = this.state.paperWidth;
+      }
+    }
+    return {
+      width: adjustedWidth,
+      height: adjustedHeight,
+    };
+  }
+
+  // renders the whole presentation area
   renderPresentationArea() {
-    let slideObj = this.props.currentSlide;
+    // sometimes tomcat publishes the slide url, but the actual file is not accessible (why?)
+    if (this.props.currentSlide &&
+        this.props.currentSlide.width &&
+        this.props.currentSlide.height) {
+      // to control the size of the svg wrapper manually
+      // and adjust cursor's thickness, so that svg didn't scale it automatically
+      const adjustedSizes = this.calculateSize();
```

Comment:
The problem is that it's not just mount. These calculations rely on two factors:
1. Data from received from `refs`, which is calculated in `componentDidMount` and then changed on `resize` events.
2. The existence of the slide, since it needs the slide's width and height.

If we run this function in parallel with `getPaperSizes` (changed the name to `getPresentationSizesAvailable`), then it has to be included in the resize update as well. And then there will be edge cases, like what to do when there is no slide, and the width/height from `refs` was updated, but this particular function couldn't do its job.

You have a valid point here, I also run it twice - for `renderPresentationArea` and for `renderWhiteboardToolbar`.
Need to think of a more efficient solution, but it can become messy very quickly.
If you have any propositions - let's discuss.